### ISSUE TITLE
check free surface computations with new manifold

### ISSUE THREAD

**Issue by tjhei** [geodynamics/aspect#250]
- are boundaries handled correctly?
- can we use the manifold description or do we need to fall back to straight interior cells?
- what about the mapping?

**Comment by ian-r-rose** [comment]
Yes, I have been meaning to look into this.  

I suspect that the best solution will be to use the manifold description for initial refinement, then to detach that manifold.  Is that something that is doable with the current setup? 

I hope to look into this in the next couple of days, and maybe submit a PR to this effect.  Any tips before I take a crack at it?

**Comment by bangerth** [comment]
Yes, but it's a bit more complicated because meshes are built and re-built all the time when you use adaptivity in parallel.

What you need to do is to use the signal-slot mechanism of the triangulation where you can perform certain actions immediately before and immediately after mesh refinement. I don't know what exactly it is you want to do, but it's possible. Take a look, for example, how I use these signals in my patch for the spherical shell geometry.

**Comment by tjhei** [comment]
Thanks, Ian. You should be able to set the manifold to ```FlatManifold```. Of course this needs to wait until we merge #242.

**Comment by ian-r-rose** [comment]
Of course, best case scenario is everything still works.

I can get working on it beforehand if I just branch off of that PR, though, right?

This reminds me, though.  I see that deal.ii has a new communicate_locally_moved_vertices() for parallel triangulations.  Would it be a good idea to offload some of the ALE stuff to that functionality

**Comment by bangerth** [comment]
Yes, if you can leverage deal.II functionality to do what you need to do in ASPECT, then that's always a good idea. It takes away a maintenance burden for you and if someone improves the generic infrastructure you benefit as well.

**Comment by ian-r-rose** [comment]
I think that this has been addressed by #368 and superseded by #369, so can be closed.
